cardinal/Dockerfile.debug

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.21 AS build
+
+# Build Delve
+RUN go install github.com/go-delve/delve/cmd/dlv@latest
+
+WORKDIR /usr/src/app
+
+copy . .
+copy vendor vendor/
+
+RUN go build -gcflags="all=-N -l" -v -o /usr/local/bin/app
+
+FROM ubuntu:22.04
+COPY --from=build /go/bin/dlv /usr/local/bin
+COPY --from=build /usr/local/bin/app /usr/local/bin/
+
+CMD ["dlv", "--listen=:40000", "--headless=true", "--api-version=2", "--accept-multiclient", "exec", "/usr/local/bin/app"]
```

Comment:
The change here is that "app" is now run under dlv which runs the app in debug mode and exposes a port at 40000 to attach a debug client. This is identical to the other dockerfile except for dlv. 

The other change is compilation `-gcflags="all=-N -l"` makes it so the app is compiled without optimization so that it can be debugged with dlv.